### ISSUE TITLE
Normative / non-normative language (HTTPS-25)

### ISSUE THREAD

**Issue by dlemire60** [oasis-tcs/openc2-impl-https#50]
The OASIS TAB wrote:
> Set and then use language for distinguishing normative from non-normative content

Submitted in [email](https://lists.oasis-open.org/archives/openc2-comment/201812/msg00002.html).

Details in OASIS JIRA:  [TAB-1639](https://issues.oasis-open.org/browse/TAB-1639).

**Comment by jmbrule** [comment]
The SLPF had a similar comment, but do not have a proposed resolution.  We do use keywords such as MUST, MUST NOT, SHALL, SHOULD.   What is the proposed resolution?

**Comment by dlemire60** [comment]
All three specifications had similar comments from the TAB. I asked Chet Ensign for advice and got the following:
>In any case one way I see it done is that the section title includes (non-normative) if the content is *not* - e.g. "1.3 Example (non-normative)"
>
> Another way is to include a sentence at the first part of the section. E.g. 
> 2. Types 
>(The content of this section is normative.)
>
> Yet a third approach is to declare at the top "The content of this specification is normative unless declared to be non-normative." 
> Then you just label non-normative sections. 
>
> Of the 3, I like the second approach because it is explicit. 

I agree with Chet about using option 2, although in a few instances we might need "_the content of this section is normative except where labeled otherwise_", such as when there are examples blended in with requirements. My original draft of the HTTPS spec had such headers; style-wise I'd make them normal-size, italicized text immediately following the header line, e.g., 

**2.0 System Architecture**
_This section is non-normative._

**Comment by jmbrule** [comment]
Thank you.  I have a ‘slight’ preference for option two with the addition of “"the content of this section is normative except where labeled otherwise"” where appropriate.
Note the word ‘slight’.  Just select whatever and let me know so I can get chomping on the SLPF.

VR

Joe Brule

**Comment by dlemire60** [comment]
I went looking at some recently announced committee specifications to see how they dealt with identifying normative / non-normative content, per the TAB comments we received on all three specifications. What I see:
1) The Universal Business Language OASIS Standard appends appropriate section headers with "(Non-normative)"
2) The OSLC Architecture Management committee specs tag non-normative sections only with _This section is non-normative._ immediately after the section header
3) The TOSCA committee spec embeds words about normative or non- into the intro text for each major section (1.0, 2.0), for example "Except for the examples, this section is *normative* and ..."

Chet Ensign indicate a preference for approach #2, which I also like. Based on the examples I've looked at, it seems to be acceptable to identify just the non-normative sections, leaving everything else as implicitly normative. I'm going to create a pull request for the HTTPS spec using this approach.
